### PR TITLE
initial documentation of currency-utils

### PR DESCRIPTION
**Pull request by Aierie**
The goal of this PR is to understand what currency utilities do, and communicate that understanding in docs. As part of that, I also separated the currency utilities into several smaller files.

### Some terms I came across and am not too sure about (referenced in camelCase in functions)
native amount - the amount in the native currency
native currency - the currency that you select in Rainbow in your global settings
native display - how users are conventionally shown an amount in the specified currency
price unit - (unsure) the amount needed to convert a given currency to native currency
asset - refers to the data structure that is used by Rainbow to specify an on-chain asset, functions here typically only use the symbol and decimals properties of assets

### CODE REVIEW

File: packages/cardpay-sdk/sdk/currency-utils/other.ts [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+export const formatCurrencyAmount = (amount: BigNumberish, decimalPlaces = 2) => {
+  return new BigNumber(amount).toFormat(decimalPlaces);
+};
+
+/**
+ * Returns a promise that resolves after `ms` milliseconds
+ */
+export const delay = (ms: number): Promise<void> => {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+};
```
Comment (Aierie) [inline packages/cardpay-sdk/sdk/currency-utils/other.ts:173]: Seems like this doesn't really belong in `currency-utils`

File: packages/cardpay-sdk/sdk/currency-utils/type-conversion.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+/**
+ * Converts a string to a number
+ */
+export const convertStringToNumber = (value: BigNumberish) => new BigNumber(value).toNumber();
+
+/**
+ * Converts `value` to a string. Can return `NaN`.
+ */
+export const convertNumberToString = (value: BigNumberish): string => new BigNumber(value).toFixed();
```
Comment (Aierie) [inline packages/cardpay-sdk/sdk/currency-utils/type-conversion.ts:26]: Functions in this file do not have great names. Really they just typecast from whatever is provided to the type they specify.

Comment (Aierie) [inline packages/cardpay-sdk/sdk/currency-utils/type-conversion.ts:26]: Functions in this file are misleadingly named. They really just convert any valid `BigNumberish` to the specified type.

File: packages/cardpay-sdk/sdk/currency-utils/formatting-plus-plus.ts [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+export const formatCurrencyAmount = (amount: BigNumberish, decimalPlaces = 2) => {
+  return new BigNumber(amount).toFormat(decimalPlaces);
+};
+
+/**
+ * Returns a promise that resolves after `ms` milliseconds
+ */
+export const delay = (ms: number): Promise<void> => {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+};
```
Comment (Aierie) [inline packages/cardpay-sdk/sdk/currency-utils/formatting-plus-plus.ts:173]: I hope we can move this out of `currency-utils` someday!

File: packages/cardpay-sdk/sdk/currency-utils/type-conversion.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+/**
+ * Converts `value` to a number
+ */
+export const convertStringToNumber = (value: BigNumberish) => new BigNumber(value).toNumber();
+
+/**
+ * Converts `value` to a string.
+ */
+export const convertNumberToString = (value: BigNumberish): string => new BigNumber(value).toFixed();
```
Comment (Aierie) [inline packages/cardpay-sdk/sdk/currency-utils/type-conversion.ts:26]: Functions in this file are currently misleadingly named about what they accept

**Comment by topwebtek7** [comment]
@Aierie great job in this PR! This change will be really helpful understanding existing currency utils.
just a request - existing functions might be already being used and could you list breaking change utils list in pr description like function names changed?

**Comment by Aierie** [comment]
I avoided changing function definitions or names so there should be no breaking changes, but will double check that nothing was changed.

File: packages/cardpay-sdk/sdk/currency-utils/types.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+import BigNumber from 'bignumber.js';
+
+/**
+ * Arguments acceptable to a BigNumber constructor, currently equivalent to BigNumber.Value
+ */
+export type BigNumberish = BigNumber.Value;
```
Comment (Aierie) [inline packages/cardpay-sdk/sdk/currency-utils/types.ts:6]: @topwebtek7 this was the only thing changed. `BigNumberish` was `string | number | BigNumber`. The definition of `BigNumber.Value`:

https://github.com/MikeMcl/bignumber.js/blob/ae9cecab22a105c063308dc96bdb37f1bffed976/bignumber.d.ts#L327

Comment (Aierie) [reply packages/cardpay-sdk/sdk/currency-utils/types.ts:6]: Figured that since BigNumber has good types, we should be using them

**Comment by Aierie** [comment]
> ### Some terms I came across and am not too sure about (referenced in camelCase in functions)
> native amount - the amount in the native currency
native currency - the currency that you select in Rainbow in your global settings
native display - how users are conventionally shown an amount in the specified currency
price unit - (unsure) the amount needed to convert a given currency to native currency
asset - refers to the data structure that is used by Rainbow to specify an on-chain asset, functions here typically only use the symbol and decimals properties of assets

Interested in whether these terms make sense as well, from your understanding(s)!

**Review by topwebtek7 — APPROVED** [review]
LGTM! 🚢

**Pull request merged**